vmm/src/igvm/igvm_loader.rs

```diff
@@ -0,0 +1,392 @@
+                loaded_info.snp_id_block.reserved[..3].copy_from_slice(reserved);
+                loaded_info.snp_id_block.ld[..48].copy_from_slice(ld);
+                loaded_info.snp_id_block.family_id[..16].copy_from_slice(family_id);
+                loaded_info.snp_id_block.image_id[..16].copy_from_slice(image_id);
+                loaded_info.snp_id_block.version = *version;
+                loaded_info.snp_id_block.guest_svn = *guest_svn;
+                loaded_info.snp_id_block.id_key_algorithm = *id_key_algorithm;
+                loaded_info.snp_id_block.author_key_algorithm = *author_key_algorithm;
+                loaded_info.snp_id_block.id_key_signature = **id_key_signature;
+                loaded_info.snp_id_block.id_public_key = **id_public_key;
+                loaded_info.snp_id_block.author_key_signature = **author_key_signature;
+                loaded_info.snp_id_block.author_public_key = **author_public_key;
+            }
+            IgvmDirectiveHeader::X64VbsVpContext {
+                vtl: _,
+                registers: _,
+                compatibility_mask: _,
+            } => {
+                todo!("VbsVpContext not supported");
+            }
+            IgvmDirectiveHeader::VbsMeasurement { .. } => {
+                todo!("VbsMeasurement not supported")
+            }
+            IgvmDirectiveHeader::ParameterInsert(IGVM_VHS_PARAMETER_INSERT {
+                gpa,
+                compatibility_mask: _,
+                parameter_area_index,
+            }) => {
+                debug_assert!(gpa % HV_PAGE_SIZE == 0);
+
+                let area = parameter_areas
+                    .get_mut(parameter_area_index)
+                    .expect("igvmfile should be valid");
+                match area {
+                    ParameterAreaState::Allocated { data, max_size } => loader
+                        .import_pages(
+                            gpa / HV_PAGE_SIZE,
+                            *max_size / HV_PAGE_SIZE,
+                            BootPageAcceptance::ExclusiveUnmeasured,
+                            data,
+                        )
+                        .map_err(Error::Loader)?,
+                    ParameterAreaState::Inserted => panic!("igvmfile is invalid, multiple insert"),
+                }
+                *area = ParameterAreaState::Inserted;
+                gpas.push(GpaPages {
+                    gpa: *gpa,
+                    page_type: hv_isolated_page_type_HV_ISOLATED_PAGE_TYPE_UNMEASURED,
+                    page_size: hv_isolated_page_size_HV_ISOLATED_PAGE_SIZE_4KB,
+                });
+            }
+            IgvmDirectiveHeader::ErrorRange { .. } => {
+                todo!("Error Range not supported")
+            }
+            _ => {
+                todo!("Header not supported!!")
+            }
+        }
+    }
+    loaded_info.length = loader.gets_total_bytes_written();
+
+    debug!("Loaded info xcr0: {:0x}", loaded_info.vmsa.xcr0);
```

Comment:
I think it would be ideal to dump the complete VMSA page information for debugging purposes.